pdns/recursordist/test-syncres_cc2.cc

```diff
@@ -62,6 +63,22 @@ BOOST_AUTO_TEST_CASE(test_referral_depth)
     BOOST_CHECK(false);
   }
   catch (const ImmediateServFailException& e) {
+    BOOST_CHECK(e.reason.find("max-recursion-depth") != string::npos);
+  }
+
+  // Then check if the setup with high limit is OK.
```

Comment:
It doesn't seem right to do a new test without clearing all the caches (records, EDNS statuses, etc..) which would be painful, perhaps this should be a separate test case instead, something like `BOOST_AUTO_TEST_CASE(test_referral_depth_ok)`?

Reply:
I hit it in the (fixed) positive test case.